### PR TITLE
Add hint option to Checkbox input

### PR DESCRIPTION
**Pull request by kat3samsin**
## Description

Add hint on checkbox validation.

image

## Checklist

- [x] This PR has good automated test coverage
- [x] The storybook for the component has been updated

## Steps to Test

1. Pull down PR.
2. `npm run dev`.
3. Open [checkbox](https://deploy-preview-377--vip-design-system-components.netlify.app/?path=/docs/form-checkbox--docs) component

### CODE REVIEW

File: src/system/Form/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -26,21 +33,53 @@ const Checkbox = ( {
...
+						<StyledCheckbox
+							onCheckedChange={ disabled ? undefined : onCheckedChange }
+							aria-disabled={ disabled }
+							aria-labelledby={ `label-${ forLabel }` }
+							aria-describedby={ hasError ? `describe-${ forLabel }-validation` : undefined }
+							variant={ variant }
+							{ ...props }
+						>
+							<StyledIndicator variant={ variant } />
+						</StyledCheckbox>
```
Comment (djalmaaraujo) [inline src/system/Form/Checkbox/Checkbox.tsx:61]: This whole block is repeated below.

File: src/system/Form/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -26,21 +33,53 @@ const Checkbox = ( {
...
+				</React.Fragment>
+			) : (
+				<StyledCheckbox
+					onCheckedChange={ disabled ? undefined : onCheckedChange }
+					aria-disabled={ disabled }
+					variant={ variant }
+					{ ...props }
+				>
+					<StyledIndicator variant={ variant } />
+				</StyledCheckbox>
```
Comment (djalmaaraujo) [inline src/system/Form/Checkbox/Checkbox.tsx:80]: Repeat

File: src/system/Form/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -26,21 +33,53 @@ const Checkbox = ( {
...
+						>
+							<StyledIndicator variant={ variant } />
+						</StyledCheckbox>
+						{ label && (
+							<Label required={ required } htmlFor={ forLabel } id={ `label-${ forLabel }` }>
+								{ label }
+							</Label>
+						) }
+					</Flex>
+					<Validation isValid={ false } describedId={ forLabel }>
```
Comment (djalmaaraujo) [inline src/system/Form/Checkbox/Checkbox.tsx:68]: isValid is always false?

**Review by djalmaaraujo — CHANGES_REQUESTED** [review]
@kat3samsin I believe this PR is doing way more than just adding the hint to checkboxes. This changes the way the checkbox is made.

It does not feel "right" the Checkbox should handle this, but a component more like "FormField" that would use checkbox/radio/input, add a label, validation, etc.

Doing the way this PR is implemented we would have code duplication with other components.

Should we discuss more in depth?

**Comment by djalmaaraujo** [comment]
@kat3samsin This "works", might just need CSS, can you check?

```
<Form.Input
			forLabel="input-with-error"
			label="Error Checkbox"
			errorMessage="Please type numeric characters only"
			as={ () => (
				<Checkbox onCheckedChange={ () => {} } checked aria-labelledby="input-with-error-var" />
			) }
			hasError
		/>
```

image

**Comment by kat3samsin** [comment]
>This "works", might just need CSS, can you check?
```
<Form.Input
			forLabel="input-with-error"
			label="Error Checkbox"
			errorMessage="Please type numeric characters only"
			as={ () => (
				<Checkbox onCheckedChange={ () => {} } checked aria-labelledby="input-with-error-var" />
			) }
			hasError
		/>
```

Thanks for this suggestion I'll try it this way first.

**Comment by kat3samsin** [comment]
Closing this PR, I'll just go with adding <Validation> component in the UI.

**Pull request closed without merging**